pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -243,10 +243,6 @@ def get_mp_spawn_kwargs(self, trainer: Optional["pl.Trainer"] = None) -> Dict[st
         }
 
     def spawn(self, function: Callable, *args: Any, **kwargs: Any) -> Optional[Union[Any, "_SpawnOutput"]]:
-        # TODO: this todo is unclear, does it still apply?
-        # todo: precision pluging is call in accelerator setup and should be moved
-        if "XLA_USE_BF16" in os.environ:
-            del os.environ["XLA_USE_BF16"]
```

Comment:
@kaushikb11 is this something that should be set up and torn down per run? https://github.com/PyTorchLightning/pytorch-lightning/blob/aeb0b5595fd73d086f4ae0f99d3f1f112f6a4c29/pytorch_lightning/plugins/precision/tpu_bf16.py#L31

otherwise doesn't this break?
```py

t16 = Trainer(accelerator="tpu", precision="bf16")
t16.fit(model)
t = Trainer(accelerator="tpu")
t.fit(model)
```
since the latter could inadvertently re-use the environment variable?

Reply:
Agreed, it is a bug. This definitely needs to be deleted per run.

Reply:
this could be done with the `setup_environment` + `teardown` methods of the training type plugin